homeassistant/components/plugwise/climate.py

```diff
@@ -127,12 +154,21 @@ def extra_state_attributes(self) -> Mapping[str, Any] | None:
     @plugwise_command
     async def async_set_temperature(self, **kwargs: Any) -> None:
         """Set new target temperature."""
-        if ((temperature := kwargs.get(ATTR_TEMPERATURE)) is None) or not (
-            self._attr_min_temp <= temperature <= self._attr_max_temp
-        ):
-            raise ValueError("Invalid temperature change requested")
-
-        await self.coordinator.api.set_temperature(self.device["location"], temperature)
+        data: dict[str, Any] = {}
+        if ATTR_TEMPERATURE in kwargs:
+            data["setpoint"] = kwargs.get(ATTR_TEMPERATURE)
+        if ATTR_TARGET_TEMP_HIGH in kwargs:
+            data["setpoint_high"] = kwargs.get(ATTR_TARGET_TEMP_HIGH)
+        if ATTR_TARGET_TEMP_LOW in kwargs:
+            data["setpoint_low"] = kwargs.get(ATTR_TARGET_TEMP_LOW)
+
+        for _, temperature in data.items():
```

Comment:
```suggestion
        for temperature in data.values():
```

Reply:
I had found that it should be: `for temperature in list(data.values():`
I will check if it works without `list()`.